.github/workflows/e2e-test.yml

```
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2025 Canonical Ltd.
+name: E2E tests
+
+on:
+  workflow_call:
+    inputs:
+      branch_name:
+        description: "Name of the branch to checkout"
+        required: false
+        type: string
+        default: ${{ github.ref }}
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    env:
+      ACTIONS_RUNNER_DEBUG: true
+      ACTIONS_STEP_DEBUG: true
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Update and install dependencies
+        run: |
+          sudo apt update
+          sudo apt-get install -y conntrack socat ipset ebtables ansible-core
+          sudo apt install -y sshpass python3-venv pipx make git
+          pipx install --include-deps ansible
+          pipx ensurepath
+          source ~/.bashrc
+
+      - name: Generate SSH Key
+        run: |
+          ssh-keygen -t rsa -b 2048 -f ~/.ssh/id_rsa -N "" -q
+          cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+
+      - name: Clone aether-onramp repository
+        run: |
+          git clone --recursive https://github.com/opennetworkinglab/aether-onramp.git
+
+      - name: Update files
+        run: |
+          IP_ADDR=$(ip a show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
```

Here is assumed that the default interface is always `eth0`. I think this is not necessarily always correct. What about the following change?
```suggestion
          INTERFACE=$(ip route | grep default | awk '{print $5}')
          IP_ADDR=$(ip a show ${INTERFACE} | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
```

---

OK, that's fine.